c7n/resources/glue.py

```diff
@@ -413,6 +414,49 @@ class resource_type(TypeInfo):
         cfn_type = 'AWS::Glue::SecurityConfiguration'
 
 
+@GlueSecurityConfiguration.filter_registry.register('kms-key')
+class KmsFilter(KmsRelatedFilter):
+    """
+    Filter a resource by its associcated kms key and optionally the aliasname
+    of the kms key by using 'c7n:AliasName'
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: glue-security-configuration-kms-key
+            resource: glue-security-configuration
+            filters:
+              - type: kms-key
+                key: c7n:AliasName
+                value: "^(alias/aws/)"
+                op: regex
+    """
+    schema = type_schema(
+        'kms-key',
+        key_type={'type': 'string', 'enum': [
+            's3', 'cloudwatch', 'job-bookmarks']},
+        rinherit=ValueFilter.schema,
+        **{'match-resource': {'type': 'boolean'},
+           'operator': {'enum': ['and', 'or']}})
+
+    RelatedIdsExpression = ''
+    key_type = ''
+
+    def __init__(self, data, manager=None):
+        super(KmsFilter, self).__init__(data, manager)
+        self.key_type = self.data.get('key_type', '')
+        if self.key_type == 's3':
```

Comment:
instead of having multiple if statements, can we create a mapping like {'s3': 'EncryptionConfiguration.S3Encryption[].KmsKeyArn'}

Reply:
Yep makes sense